PythonPonies/bangazon-api
# Adds model, serializer and relevant code for the Products table

# Description
This pull request resolved merge conflicts with code on master and adds:
1. The model for Products
1. The serializer for Products
1. The url patterns for Product Details and Product List
1. I also converted any comments within classes and methods to docstring notation

## Number of Fixes
6

## Related Ticket(s)
Client developers can access resources via an API #1
Create Product Model #52
Create `Product Details` View #44
Create `Products List` View #33
Route the model `products-list` to its view #26 
Route the model `products` to its view #25
Create Products Serializer #15 

## Problem to Solve
Resolve merge conflicts and add functionality to the API so that users can access products details and a list of products via JSON.

## Proposed Changes
Adds model, serializer and relevant code for the Products table.

## Expected Behavior
A user can access all products information.

## Steps to Test Solution
To test: 

1. Git fetch this branch
1. Migrate and run the django server.
1. Go to `http://127.0.0.1:8000/products/` to view a list of products
1. Go to `http://127.0.0.1:8000/products/1` to see a product detail view.
1. You can test adding a product from here.

## Testing

[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Documentation
[ ] There is new documentation in this pull request that must be reviewed.
[x] I added documentation for any new classes/methods

## Deploy Notes
none
